src/xmipp/libraries/reconstruction/angular_continuous_assign2.cpp

```diff
@@ -607,8 +634,13 @@ void ProgAngularContinuousAssign2::processImage(const FileName &fnImg, const Fil
     if (hasCTF)
     {
     	rowOut.setValue(MDL_CTF_DEFOCUSU,old_defocusU+p(10));
-    	rowOut.setValue(MDL_CTF_DEFOCUSV,old_defocusV+p(11));
+		if (sameDefocus)
+    		rowOut.setValue(MDL_CTF_DEFOCUSV,old_defocusV+p(10)); 
+		else
+			rowOut.setValue(MDL_CTF_DEFOCUSV,old_defocusV+p(11));
     	rowOut.setValue(MDL_CTF_DEFOCUS_ANGLE,old_defocusAngle+p(12));
+		if (sameDefocus)
+    		rowOut.setValue(MDL_CTF_DEFOCUS_CHANGE,0.5*(p(10)+p(11)));
```

Comment:
In the if it should be if !sameDefocus, and the case for sameDefocus is undefined